plugins/inventory/robot.py

```diff
@@ -136,24 +136,27 @@ def populate(self, servers):
         for server in servers:
             s = server['server']
             server_name = s.get('server_name') or s['server_ip']
-            server_name = '{0}-{1}'.format(server_name, s['server_number'])
-
             matched = self.filter(s, filters)
             if matched:
-                self.inventory.add_host(server_name)
-                self.inventory.set_variable(server_name, 'ansible_host', s['server_ip'])
-                for hostvar, hostval in s.items():
-                    self.inventory.set_variable(server_name, "{0}_{1}".format('hrobot', hostvar), hostval)
-
-                # Composed variables
-                server_vars = self.inventory.get_host(server_name).get_vars()
-                self._set_composite_vars(self.get_option('compose'), server_vars, server_name, strict=strict)
-
-                # Complex groups based on jinja2 conditionals, hosts that meet the conditional are added to group
-                self._add_host_to_composed_groups(self.get_option('groups'), server, server_name, strict=strict)
-
-                # Create groups based on variable values and add the corresponding hosts to it
-                self._add_host_to_keyed_groups(self.get_option('keyed_groups'), server, server_name, strict=strict)
+                if self.inventory.get_host(server_name) is None:
```

Comment:
I don't think that's a good idea. People might want to combine different inventory sources, so the warning should happen if this plugin uses a hostname twice, not if the host was already defined by another plugin.

Reply:
is there a way to figure out this? Like, `plugin uses a hostname twice`?

Reply:
but if someone, uses two inventory sources, and 
one inventory sources has been added hostname with server-1, and hetzner robot inventory sources which find item server-1 one will override it?

Reply:
You can simply create a `set` where you add every name you find, and if it already contained a name before adding, you have a duplicate.

Reply:
And they won't override, only if they define exactly the same variables.

Reply:
ah, got it. fixed

Reply:
> You can simply create a `set` where you add every name you find, and if it already contained a name before adding, you have a duplicate.

:) no ansible magic though